sound/soc/sof/intel/hda-codec.c

```diff
@@ -171,20 +170,20 @@ static int hda_codec_probe(struct snd_sof_dev *sdev, int address,
 	if (!hda_priv)
 		return -ENOMEM;
 
-	hda_priv->codec.bus = hbus;
-	hdev = &hda_priv->codec.core;
-	codec = &hda_priv->codec;
-
 	/* only probe ASoC codec drivers for HDAC-HDMI */
 	if (!hda_codec_use_common_hdmi && (resp & 0xFFFF0000) == IDISP_VID_INTEL)
 		type = HDA_DEV_ASOC;
 
-	ret = snd_hdac_ext_bus_device_init(&hbus->core, address, hdev, type);
+	codec = hda_codec_device_init(&hbus->core, address);
```

Comment:
Hmm, not sure if the type setting goes correctly here. It seems the assignment on L175 is no longer used and variable "type" goes unused after this change.

In SOF, we basicly only use HDA_DEV_ASOC to support one old configuration (some GLK chromebooks using hdac_hdmi), but in 97% of the configurations, init should be done with HDA_DEV_LEGACY (and use hdac_hda.c).

Reply:
Good catch. This is probably the cause for the CI failing. Updated `hda_codec_device_init()` to accept additional 'type' argument to address the problem. For skylake-driver it's easier - it was always assigning `HDA_DEV_ASOC` upfront.